### PR TITLE
[Gogoanime Presence] code cleanup + domain check for clone + added Watch button

### PR DESCRIPTION
**Pull request by PsychoPast**
image
image

### CODE REVIEW

File: websites/G/Gogoanime/dist/metadata.json [hunk 1]
```diff
@@ -1,23 +1,26 @@
 {
   "author": {
-    "name": "PsychoPast",
+    "name": "Psycho",
     "id": "444779014888882177"
   },
+  "contributors": [{
+    "name": "Undo",
+    "id": "246660882803720193"
+  }],
```
Comment (Bas950) [inline websites/G/Gogoanime/dist/metadata.json:9]: didnt undo want to be the author?

File: websites/G/Gogoanime/dist/metadata.json [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
   "service": "Gogoanime",
   "description": {
-    "en": "Watch anime online in English. You can watch free series and movies online with English subtitles."
+    "en": "Watch anime online in English. You can watch free series and movies online and English subtitle. Free download high quality anime."
   },
-  "url": [
-    "gogoanime.vc",
-    "gogoanime.ai"
-  ],
```
Comment (Bas950) [inline websites/G/Gogoanime/dist/metadata.json:13]: you still need atleast 1 url (as fallback for if for some reason the regex fails)

File: websites/G/Gogoanime/presence.ts [hunk 3]
```diff
@@ -1,120 +1,253 @@
...
+  },
+  date: {
+    data: number;
+  }
+}
+
+const presence = new Presence({
+  clientId: "696341580096733185",
+  injectOnComplete: false,
+  appMode: true
```
Comment (Bas950) [inline websites/G/Gogoanime/presence.ts:23]: injectOnComplete is false by default so no need to set.

File: websites/G/Gogoanime/presence.ts [hunk 4]
```diff
@@ -1,120 +1,253 @@
...
-  }
-});
+presence.on("UpdateData", async () => {
+  if (!isDomainChecked) {
+    isDomainChecked = true;
+    await checkDomain().then(result => {
+      isClone = result; 
+      console.log(
+        `[Premid Gogoganime Presence] The following gogoanime domain is ${result ? "a clone therefore not supported" : "supported"} by this extension.`);
+      });
```
Comment (Bas950) [inline websites/G/Gogoanime/presence.ts:136]: use presence.info or presence.error (and you spelled PreMiD wrong :triggered:)

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/G/Gogoanime/presence.ts (hunk 4 above)
Comment (PsychoPast) [reply websites/G/Gogoanime/presence.ts]: Annnd this is why I should read the docs more often xd Thx

File: websites/G/Gogoanime/dist/metadata.json (hunk 1 above)
Comment (PsychoPast) [reply websites/G/Gogoanime/dist/metadata.json:9]: Not really. I worked closely with them to implement the domain check and they're fine with it

File: websites/G/Gogoanime/dist/metadata.json [hunk 5]
```diff
@@ -1,23 +1,38 @@
...
+    "gogoanime.in",
+    "gogoanime.video",
+    "gogoanime.ai",
+    "gogoanime.vc"
   ],
-  "version": "1.0.9",
-  "logo": "https://i.imgur.com/Awp1MUc.png",
+  "regExp":"[0-9w]{0,6}.*gogoanime[s]*.[a-z]+",
+  "version": "1.1.0",
+  "logo": "https://i.imgur.com/OnaEYdo.png",
```
Comment (Bas950) [inline websites/G/Gogoanime/dist/metadata.json:28]: please do v2.0.0

File: websites/G/Gogoanime/presence.ts [hunk 6]
```diff
@@ -1,120 +1,257 @@
-var presence = new Presence({
-  clientId: "696341580096733185"
+type domainList = string[];
+
```
Comment (Bas950) [inline websites/G/Gogoanime/presence.ts:2]: kinda useless type :/

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**